### ISSUE TITLE
[BUG] Processing is only stopped for duplicates after applying other plugins

### ISSUE THREAD

**Issue by thomwiggers** [paperless-ngx/paperless-ngx#7982]
### Description

Currently, Paperless first applies plugins that might be very expensive to run, before it realizes that the document is actually a duplicate.

This seems to be dictated by the following lines:

https://github.com/paperless-ngx/paperless-ngx/blob/0a4da6db554ce945a8d3ec0564a5f9db9cc2e751/src/documents/tasks.py#L114-L119

I suggest separating the duplicates detection into a separate Plugin and inserting that in the queue before as many expensive operations as possible. 

### Steps to reproduce

1. Re-upload 200MB vector PDFs full of building drawings
2. Only after several minutes of processing per file does Paperless realize that it just wasted a ton of time

### Webserver logs

```bash
[2024-10-22 16:01:01,892] [DEBUG] [paperless.tasks] Skipping plugin CollatePlugin
[2024-10-22 16:01:01,893] [DEBUG] [paperless.tasks] Executing plugin BarcodePlugin
[2024-10-22 16:01:01,893] [DEBUG] [paperless.barcodes] Scanning for barcodes using ZXING
[2024-10-22 16:01:01,896] [DEBUG] [paperless.barcodes] PDF has 1 pages
[2024-10-22 16:01:01,896] [DEBUG] [paperless.barcodes] Processing page 0
[2024-10-22 16:01:18,239] [DEBUG] [paperless.barcodes] Image is at /tmp/paperless/tmp9l_c5_cc/barcodeyyr0uvqy/1a2e25e8-775a-4d82-aaf2-ff9c7324139d-1.ppm
[2024-10-22 16:01:19,292] [DEBUG] [paperless.barcodes] Scanning for barcodes using ZXING
[2024-10-22 16:01:19,294] [DEBUG] [paperless.barcodes] PDF has 1 pages
[2024-10-22 16:01:19,294] [DEBUG] [paperless.barcodes] Processing page 0
[2024-10-22 16:01:35,464] [DEBUG] [paperless.barcodes] Image is at /tmp/paperless/tmp9l_c5_cc/barcodeyyr0uvqy/0d205bb5-a870-42e5-9f85-5537fb12fa63-1.ppm
[2024-10-22 16:01:36,551] [INFO] [paperless.tasks] BarcodePlugin completed with no message
[2024-10-22 16:01:36,552] [DEBUG] [paperless.tasks] Executing plugin WorkflowTriggerPlugin
[2024-10-22 16:01:36,556] [INFO] [paperless.tasks] WorkflowTriggerPlugin completed with:
[2024-10-22 16:01:36,556] [DEBUG] [paperless.tasks] Executing plugin ConsumeTaskPlugin
[2024-10-22 16:01:36,601] [ERROR] [paperless.consumer] Not consuming X.pdf: It is a duplicate of X- (#823).
```

### Paperless-ngx version

2.12.2

### Host OS

N/A

### Installation method

Docker - official image

### Browser

N/A

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

**Comment by shamoon** [comment]
A suggestion is clearly not a bug. You can open a request if you’d like

**Comment by stumpylog** [comment]
With a little thought, it should be clear why things run in the order they do.  This won't be changing

**Comment by thomwiggers** [comment]
None of the processing plugins will change the MD5-sum of the raw input file, so putting that check before expensive processes can be easily done.

**Comment by stumpylog** [comment]
So splitting a file at a barcode doesn't change the file?

**Comment by thomwiggers** [comment]
Ok, I forgot about the splitting function. But, the way it's implemented, it actually doesn't change the raw input file: it spawns new tasks and stops consuming the original. I'd argue that the time/processing savings are worth the edge case that someone resubmits a file with splitting barcodes that was already processed while barcode splitting was disabled.
